vars/dockerExecute.groovy

```diff
@@ -379,6 +381,18 @@ def securityContextFromOptions(dockerOptions) {
     return securityContext
 }
 
+/*
+ * Picks the first volumeBind option
+ * and translates it into containerMountPath
+ */
+@NonCPS
+def containerMountPathFromVolumeBind(dockerVolumeBind) {
+    if (dockerVolumeBind) {
+        return dockerVolumeBind[0].split(":")[1]
```

Comment:
Are you sure all customers will use `:` as separator?
Docker also provides the option to specify those via `source` and `target` properties. For those use cases your function is not going to work.

Reply:
Also why only the **first** volume get's translated?

Reply:
> Also why only the **first** volume get's translated?

It is `containerMountPath`. The user cannot influence WHAT is mounted, only to what path.

Reply:
> Are you sure all customers will use `:` as separator? Docker also provides the option to specify mounts via `source` and `target` properties. For those use cases your function is not going to work. Not sure if this is supported by Piper.

Do you mean [this](https://docs.docker.com/storage/bind-mounts/#differences-between--v-and---mount-behavior)? This ist actually a `mount` command and not `bind`. So I do not think it is supported.

Reply:
As part of the JaaS setup within SAP, we do not permit the use of hostPath volumes. This practice aligns with security best practices, especially in a shared cluster environment. The proposed change would have an impact on our internal customers

Reply:
> As part of the JaaS setup within SAP, we do not permit the use of hostPath volumes. This practice aligns with security best practices, especially in a shared cluster environment. The proposed change would have an impact on our internal customers

Does it? This pull request is about volume mounts, it does not touch on what is or can be mounted. @basava12345 TBH, I didn't really get your point. Maybe you can give an example.

Reply:
The volume is still hard coded to an emptydir ([code](https://github.com/SAP/jenkins-library/blob/8175b2d456f3ae99fe73fb4a9c9fc60ede9193e7/vars/dockerExecuteOnKubernetes.groovy#L385))

Reply:
By seeing docker volume binding, i assumed that usage of hostpath volume is coming into picture. You can ignore the  comment if its emptyDir volume.

Reply:
@basava12345 , @c0d1ngm0nk3y  , the [code](https://github.com/SAP/jenkins-library/blob/8175b2d456f3ae99fe73fb4a9c9fc60ede9193e7/vars/dockerExecuteOnKubernetes.groovy#L3850) only talks about the volume mounts in the main app container running the pod and this functionality is unchanged, if i understand correctly the [change](https://github.com/SAP/jenkins-library/blob/8175b2d456f3ae99fe73fb4a9c9fc60ede9193e7/vars/dockerExecuteOnKubernetes.groovy#L482-L483) can allow any path to be mounted in the init container running the pod . 

Since the environment is shared does that allow me to bring a path from any other project into my running pod with the init container ?

Reply:
> @basava12345 , @c0d1ngm0nk3y , the [code](https://github.com/SAP/jenkins-library/blob/8175b2d456f3ae99fe73fb4a9c9fc60ede9193e7/vars/dockerExecuteOnKubernetes.groovy#L3850) only talks about the volume mounts in the main app container running in the pod and this functionality is unchanged, 

> if i understand correctly the [change](https://github.com/SAP/jenkins-library/blob/8175b2d456f3ae99fe73fb4a9c9fc60ede9193e7/vars/dockerExecuteOnKubernetes.groovy#L588-L589) can allow any path to be mounted in the side care container ?.

Yes, the path in the container can be anything, but that is only the path within the container. *WHAT* to mount is hardcoded (here: "volume").

> Since the environment is shared does that allow me to bring a path from any other project into my running pod with the side car container 

I don't se how...

> or is it a named volume that already exists which i want to mount at a specefic path in my container ?

Yes. In case of Kubernetes, it is a `emptyDir`, I do not know what the other orchestrators are doing tbh.

> if its an already existing volume who creates it ?

Hopefully, the orchestrator specific code. Or `docker` when starting it with the correct arguments.

> maybe i missing something here and so want to be sure,

I am not 100% sure that we are talking about the same thing, similar to [this comment](https://github.com/SAP/jenkins-library/pull/4673#discussion_r1399144079).

Reply:
i think it will confusing for users with the current paramter. if someone called `dockerExecute` with `dockerVolumeBind` then there is a implicit understanding in code that the target path is taken into account and an emptyDir is mounted into the target path . and this changes the understanding of a traditional volume bind `hostPath:targetPath` which is not allowed in shared k8s cluster environment but can work outside a shared k8s cluster

we have a clean parameter in `dockerExecuteOnKubernetes`, couldn't we reuse that as a new param for `dockerExecute` ? so the behavior remains the same ?

    /**
     * The path to which a volume should be mounted to. This volume will be available at the same
     * mount path in each container of the provided containerMap. The volume is of type emptyDir
     * and has the name 'volume'. With the additionalPodProperties parameter one can for example
     * use this volume in an initContainer.
     */
    'containerMountPath',
     /**

Reply:
To summarise the meeting we had:

We will stick to `volumeBind`, but make sure that the `Name` is `volume`. With that, we would still be flexible to enhance in the future, but for the user there won't be surprises what ist actually mounted. We should document this restriction though.